### PR TITLE
[Variation] Introduce code for option values and replace presentation with name 

### PR DESCRIPTION
**Pull request by vikey89**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This PR is based from [#4213](https://github.com/Sylius/Sylius/issues/4213)

### CODE REVIEW

**Comment by vikey89** [comment]
@pjedrzejewski I think I've finished on this PR.

**Comment by vikey89** [comment]
@pjedrzejewski @michalmarcinkowski I would like to help with something else :)
Any suggestion?

File: app/migrations/Version20160220110242.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE sylius_product_option DROP name');
+        $this->addSql('DROP INDEX fulltext_search_idx ON sylius_search_index');
```
Comment (pjedrzejewski) [inline app/migrations/Version20160220110242.php]: This and line below seem incorrect.

**Comment by pjedrzejewski** [comment]
@vikey89 Great, thank you so much for your time! Could you please squash the commits and have a look at the migration? Otherwise looks good to go!

File: src/Sylius/Component/Variation/Model/OptionValue.php [hunk 2]
```diff
@@ -105,13 +105,13 @@ public function setValue($value)
     /**
      * {@inheritdoc}
      */
-    public function getName()
+    public function getCodeOption()
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Variation/Model/OptionValue.php]: This should be ``getOptionCode`` I think, wdyt?

Comment (vikey89) [inline src/Sylius/Component/Variation/Model/OptionValue.php]: yes, you're right.

**Comment by vikey89** [comment]
@pjedrzejewski I squashed the commits and fixes migration and option code.

File: src/Sylius/Component/Variation/Model/OptionValue.php [hunk 3]
```diff
@@ -105,13 +105,13 @@ public function setValue($value)
     /**
      * {@inheritdoc}
      */
-    public function getName()
+    public function getOptionCode()
```
Comment (okwinza) [inline src/Sylius/Component/Variation/Model/OptionValue.php]: :+1: this is much better name.

**Comment by okwinza** [comment]
@pjedrzejewski as a sidenote with optionvalues now being translatable we have to be careful when displaying all option values at once otherwise we risk facing something like this(800+ queries FTW)

zomg

In light of this my #4158 quickly becomes quite important to merge IMO.
Or maybe we can introduce some caching mechanism like memcached for doctrine or something.

**Comment by nakashu** [comment]
@okwinza is this something that can be handled by setting EAGER loading on the translations and optionValues? It should be then loaded with a single query.

http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/annotations-reference.html#onetomany

**Comment by okwinza** [comment]
ping @pjedrzejewski 
this seems to be ready to merge. And i also need your thoughts on my comment above.

**Comment by vikey89** [comment]
@pjedrzejewski @michalmarcinkowski I would like to collaborate with you more often.
Do you have some issues with more priority? :)

**Comment by okwinza** [comment]
@vikey89 try to come up with some decent price slider for example :D 

On a serious note it would be nice to have it and #4285 merged already :)
Double ping @pjedrzejewski

**Comment by pjedrzejewski** [comment]
Thanks Vincenzo! :+1:

**Comment by psyray** [comment]
I think this PR was merged too fast.
Migrations are not correctly done.

What is called migration is not only **Create and drop**, in some case, like this one, migrations must be **Create, select, insert and finally drop** like this [migration] (https://github.com/Sylius/Sylius/blob/master/app/migrations/Version20141203091118.php)

I am rewriting the migration to copy all values from the option_value table in option_value_translation table because in this state, this PR break my project and certainly other project

**Comment by psyray** [comment]
Issue & PR created

**Comment by pjedrzejewski** [comment]
Hey @psyray, sorry for that. Most migrations are autogenerated right now. We want to fix that soon, as we are minimizing the changes to schema right now. All future migrations will migrate the data as well. Thanks for your contribution!

**Pull request merged**